src/Raven.Server/Documents/TombstoneCleaner.cs

```diff
@@ -348,29 +343,136 @@ public TombstonesState()
+                }
+            }
+
+            private void SetTombstoneTypes(ITombstoneAware.TombstoneType type, SubscriptionInfo subscriptionInfo, long numberOfTombstoneLeft)
+            {
+                if (subscriptionInfo.TombStoneTypes == null)
+                {
+                    subscriptionInfo.TombStoneTypes = new TombStoneTypes();
+                }
+
+                switch (type)
+                {
+                    case ITombstoneAware.TombstoneType.Documents:
+                        subscriptionInfo.TombStoneTypes.Documents = numberOfTombstoneLeft;
+                        break;
+                    case ITombstoneAware.TombstoneType.TimeSeries:
+                        subscriptionInfo.TombStoneTypes.TimeSeries = numberOfTombstoneLeft;
+                        break;
+                    case ITombstoneAware.TombstoneType.Counters:
+                        subscriptionInfo.TombStoneTypes.Counters = numberOfTombstoneLeft;
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException(nameof(type), $"Unsupported tombstone type: {type}");
+                }
+            }
 
-            public void AddPerSubscriptionInfo(string identifier, ITombstoneAware.TombstoneType type, string collection, long etag)
+            private SubscriptionInfo CreateSubscriptionInfo(ITombstoneAware subscription, LastTombstoneInfo tombstoneInfo, string collection, long remainingTombstones)
             {
-                PerSubscriptionInfo ??= new List<SubscriptionInfo>();
-                PerSubscriptionInfo.Add(new SubscriptionInfo
+                var newSubscriptionInfo = new SubscriptionInfo
                 {
-                    Identifier = identifier,
-                    Type = type,
+                    Process = subscription.TombstoneCleanerIdentifier,
+                    Identifier = tombstoneInfo.Name,
                     Collection = collection,
-                    Etag = etag
-                });
+                    Etag = tombstoneInfo.Etag,
+                    NumberOfTombstoneLeft = remainingTombstones,
+                    CleanupStatus = remainingTombstones > 0 ? CleanupStatus.Blocking : CleanupStatus.NotBlocking
+                };
+                return newSubscriptionInfo;
             }
 
             public class SubscriptionInfo
             {
+                public string Process { get; set; }
+
                 public string Identifier { get; set; }
 
-                public ITombstoneAware.TombstoneType Type { get; set; }
+                public TombStoneTypes TombStoneTypes { get; set; }
 
                 public string Collection { get; set; }
 
                 public long Etag { get; set; }
+
+                public long NumberOfTombstoneLeft { get; set; }
+
+                public CleanupStatus CleanupStatus { get; set; }
+            }
+
+            public class TombStoneTypes
```

Comment:
Maybe inherit from `IDynamicJson` and add a `ToJson` here?

Reply:
Done